legacyCore/src/main/java/com/hamdan/forzenbook/legacy/core/adapters/FeedAdapter.kt

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(getDataAt(position))
+    }
+
+    override fun getItemCount(): Int = getDataSize()
+
+    override fun getItemViewType(position: Int): Int =
+        getDataType(position)
+}
+
+abstract class FeedItemViewModel(val type: Int) {
+    companion object {
+        const val IMAGE = 0
+        const val TEXT = 1
+        const val LOADING_MORE = 2
+        // don't believe theres a need for a loading item for the TextItem and ImageItem as coil can handle the loading situation
+    }
+}
+
+class ImageFeedItemViewModel(
+    val profilePictureUri: String?,
+    val firstName: String,
+    val lastName: String,
+    val date: String,
+    val location: String,
+    val posterId: Int,
+    val imageUri: String,
+) : FeedItemViewModel(IMAGE)
+
+class TextFeedItemViewModel(
+    val profilePictureUri: String?,
+    val firstName: String,
+    val lastName: String,
+    val date: String,
+    val location: String,
+    val posterId: Int,
+    val postText: String,
+) : FeedItemViewModel(TEXT)
+
+class LoadFeedItemViewModel() : FeedItemViewModel(LOADING_MORE)
+
+abstract class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    abstract fun bind(model: FeedItemViewModel)
+}
+
+class ImageViewHolder(
+    private val binding: ImagePostBinding,
+    private val onNameClick: (Int) -> Unit
+) :
+    ViewHolder(binding.root) {
+    override fun bind(model: FeedItemViewModel) {
+        val imageModel = model as ImageFeedItemViewModel
+        binding.apply {
+            feedUserName.text = this.root.context.getString(
+                R.string.user_name,
+                imageModel.firstName,
+                imageModel.lastName
+            )
+            feedUserName.setOnClickListener { onNameClick(imageModel.posterId) }
+            feedUserDate.text = imageModel.date
+            feedUserLocation.text = imageModel.location
+            val imageLoader = ImageLoader(itemView.context)
+            val profilePictureRequest = ImageRequest.Builder(itemView.context)
+                .data(imageModel.profilePictureUri)
+                .target(
+                    onStart = {
+                        postProfileIndicator.isVisible = true
+                    },
+                    onSuccess = { image ->
+                        postProfileIndicator.isVisible = false
+                        feedProfilePicture.setImageDrawable(image)
+                        postProfilePictureHolder.isVisible = true
+                    },
+                    onError = { _ ->
+                        postProfileIndicator.isVisible = false
+                        feedProfilePicture.setImageResource((R.drawable.logo_render_full_notext))
+                        postProfilePictureHolder.isVisible = true
+                    }
+                )
+                .build()
+            val imageRequest = ImageRequest.Builder(itemView.context)
```

Comment:
Make sure this works when scrolling fast. Reyclerviews with image loading is poopy

Reply:
it looked like it was doing it alright, but only 2 sample images to test with really.